airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java

```diff
@@ -155,28 +155,59 @@ else if (hasFailed.get()) {
         outputStatus = ReplicationStatus.COMPLETED;
       }
 
+      final StreamNameToBytesEmitted streamNameToBytesEmitted = new StreamNameToBytesEmitted();
+      messageTracker.getEmittedBytesByStream().forEach((stream, bytes) -> streamNameToBytesEmitted.setAdditionalProperty(stream, bytes));
+
+      final StreamNameToRecordsEmitted streamNameToRecordsEmitted = new StreamNameToRecordsEmitted();
+      messageTracker.getEmittedRecordsByStream().forEach((stream, records) -> streamNameToRecordsEmitted.setAdditionalProperty(stream, records));
+
       final ReplicationAttemptSummary summary = new ReplicationAttemptSummary()
           .withStatus(outputStatus)
-          .withRecordsSynced(sourceMessageTracker.getRecordCount())
-          .withBytesSynced(sourceMessageTracker.getBytesCount())
+          .withRecordsSynced(messageTracker.getTotalRecordsEmitted()) // TODO (parker) remove in favor of totalRecordsEmitted
+          .withTotalRecordsEmitted(messageTracker.getTotalRecordsEmitted())
+          .withBytesSynced(messageTracker.getTotalBytesEmitted()) // TODO (parker) rename to totalBytesEmitted?
+          .withTotalStateMessagesEmitted(messageTracker.getTotalStateMessagesEmitted())
+          .withStreamNameToBytesEmitted(streamNameToBytesEmitted)
+          .withStreamNameToRecordsEmitted(streamNameToRecordsEmitted)
           .withStartTime(startTime)
           .withEndTime(System.currentTimeMillis());
 
+      final StreamNameToRecordsCommitted streamNameToRecordsCommitted = new StreamNameToRecordsCommitted();
+      if (messageTracker.getCommittedRecordsByStream().isPresent()) {
+        messageTracker.getCommittedRecordsByStream().get().forEach((stream, records) -> {
+          streamNameToRecordsCommitted.setAdditionalProperty(stream, records);
+        });
+        summary.setStreamNameToRecordsCommitted(streamNameToRecordsCommitted);
+      } else {
+        LOGGER.warn("Could not reliably determine committed records per stream, setting streamNameToRecordsCommitted to null.");
+        summary.setStreamNameToRecordsCommitted(null); // TODO (parker) handle this more explicitly?
```

Comment:
i guess technically what you have is sufficient. null == unreliable, non-null (including 0) are all valid values. because we use a `Long` here we can technically get away with it. I wouldn't say no to something more explicit but can't think of anything that's not way more verbose off the top of my head.